app/movie/[id].tsx

```diff
@@ -0,0 +1,740 @@
+      }
+    }
+
+    return []
+  }, [movie])
+
+  const castMembers = useMemo<NormalizedCastMember[]>(() => {
+    if (!Array.isArray(rawCast)) {
+      return []
+    }
+
+    const normalized: NormalizedCastMember[] = []
+
+    rawCast.forEach(member => {
+      if (!member) return
+
+      const memberObj =
+        typeof member === 'object' && member !== null ? member : {}
+      const actorObj =
+        (memberObj as Record<string, any>).actor ??
+        (memberObj as Record<string, any>).actorInfo ??
+        (memberObj as Record<string, any>).actor_data ??
+        memberObj
+
+      const nameCandidates = [
+        typeof member === 'string' ? member : undefined,
+        (actorObj as Record<string, any>).name,
+        (actorObj as Record<string, any>).fullName,
+        (actorObj as Record<string, any>).full_name,
+        (actorObj as Record<string, any>).actor_name,
+        (memberObj as Record<string, any>).actor_name,
+        (memberObj as Record<string, any>).name,
+        [
+          (actorObj as Record<string, any>).first_name ??
+            (actorObj as Record<string, any>).firstName,
+          (actorObj as Record<string, any>).last_name ??
+            (actorObj as Record<string, any>).lastName,
+        ]
+          .filter(Boolean)
+          .join(' ')
+          .trim(),
+      ]
+
+      const name = nameCandidates.find(
+        candidate =>
+          typeof candidate === 'string' && candidate.trim().length > 0
+      ) as string | undefined
+
+      if (!name) {
+        return
+      }
+
+      const idRaw =
+        (memberObj as Record<string, any>).id ??
+        (actorObj as Record<string, any>).id ??
+        (memberObj as Record<string, any>).actor_id ??
+        (actorObj as Record<string, any>).actor_id ??
+        (memberObj as Record<string, any>).cast_id ??
+        name
+
+      const characterCandidates = [
+        (memberObj as Record<string, any>).character,
+        (memberObj as Record<string, any>).character_name,
+        (memberObj as Record<string, any>).characterName,
+        (memberObj as Record<string, any>).role,
+        (memberObj as Record<string, any>).as,
+        (memberObj as Record<string, any>).personaje,
+      ]
+
+      const character = characterCandidates.find(
+        candidate =>
+          typeof candidate === 'string' && candidate.trim().length > 0
+      ) as string | undefined
+
+      const imageCandidates = [
+        (actorObj as Record<string, any>).image,
+        (actorObj as Record<string, any>).imageUrl,
+        (actorObj as Record<string, any>).image_url,
+        (actorObj as Record<string, any>).imageProfile,
+        (actorObj as Record<string, any>).profile_image,
+        (memberObj as Record<string, any>).image,
+        (memberObj as Record<string, any>).imageProfile,
+        (memberObj as Record<string, any>).profile_image,
+        (actorObj as Record<string, any>).profile_path,
+        (memberObj as Record<string, any>).profile_path,
```

Comment:
You don't need to check all of these props, just use the ones that the BE returns